File: client/src/app/checkin/checkin.service.ts
import {Injectable} from '@angular/core';
import {BaseService} from '../shared/crud/base.service';
import {Checkin} from './checkin';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {Observable} from 'rxjs';
import {Page} from '../shared/model/page';
import {CheckinFilter} from './checkin.filter';
import {CheckinDto} from './checkin-dto';

@Injectable()
export class CheckinService extends BaseService<Checkin, number> {

  constructor(http: HttpClient) {
    super(`${environment.urlApi}/checkin`, http);
  }

  filter(checkinFilter: CheckinFilter, page: number, size: number): Observable<Page<CheckinDto>> {
    return this.http.post<Page<CheckinDto>>(`${this.url}/filter?page=${page}&size=${size}`, checkinFilter);
  }

}

File: server/src/main/java/br/com/senior/server/checkin/CheckinService.java
package br.com.senior.server.checkin;

import br.com.senior.server.crud.BaseService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CheckinService extends BaseService<Checkin, Long> {

    Page<CheckinDTO> filter(CheckinFilter checkinFilter, Pageable pageable);
}
